### ISSUE TITLE
Overly broad error: Target media is not an entire storage device

### ISSUE THREAD

**Issue by kevinoid** [WoeUSB/WoeUSB-ng#90]
## Good Habit Checklist for Issue Reporting
Checkout [Mastering Markdown · GitHub Guides](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) if you needs help on the GitHub-flavored Markdown syntax.  You may remove sections and contents that don't apply.

* [x] I've searched the issue tracker and is pretty sure that there's no duplicate issue already filed
* [x] I've built the latest development snapshot using the instructions in README and verified that the issue can still be reproduced (for bug reports)

## Issue Reproduce Instructions

1. Launch WoeUSB by running `woeusb --device Win10.iso /dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72-0:0`

## Expected Behavior

No error

## Current Behavior

WoeUSB produces the following error message whenever the device path ends with a digit:

    ERROR: Target media "/dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72-0:0" is not an entire storage device!

### WoeUSB Version

v0.2.5-64-gbcb7583

### WoeUSB Source
Built from source code, commit bcb75832cabb02fea4f0f8311366051caed7160f

### Information about the Operating System
Debian GNU/Linux bookworm/sid

### Information about the Source Media
"Windows 10 2022 Update | Version 22H2" downloaded from <https://www.microsoft.com/en-us/software-download/windows10ISO>

### Information about the Target Device
[8GB Kanguru FlashBlu USB2.0 flash drive](https://web.archive.org/web/20091112115049/http://shop.kanguru.com/index.php/flash-drives/basic-flash/kanguru-flashblu)

**Comment by JonnyTech** [comment]
AFAIK, `/dev/disk/by-id/` is not supported, use `/dev/sdX` where `X` is the appropriate identifier.

**Comment by WaxyMocha** [comment]
https://github.com/WoeUSB/WoeUSB-ng/blob/master/WoeUSB/list_devices.py#L23
Yea, as @JonnyTech mentioned. Use /dev/sdX path.

**Comment by kevinoid** [comment]
> AFAIK, /dev/disk/by-id/ is not supported, use /dev/sdX where X is the appropriate identifier.

In that case, perhaps it would make sense to print an error message to that effect?

**Comment by JonnyTech** [comment]
The error you posted clearly states that the path you entered was not a storage device.

**Comment by kevinoid** [comment]
> The error you posted clearly states that the path you entered was not a storage device.

Indeed.  Which seems misleading to me, since it is (a link to) a storage device.

**Comment by kevinoid** [comment]
To clarify, since I didn't mention it in the issue, `/dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72-0:0` is a symlink to `/dev/sdb`.

**Comment by JonnyTech** [comment]
IMHO, this tool should be used by power users who understand that one must write to a device identifier rather than a symbolic link.

**Comment by kevinoid** [comment]
Why would power users assume that this tool, unlike other tools for disk management, would be unable to support symlinks `/dev/disk/by-id`?

**Comment by JonnyTech** [comment]
The author of the code already pointed you to the location in the source where it is stated.

What is the output of `ls -la /dev/disk/by-id/`? Does it have the `-0:0` suffix at the end? I have not seen that format before. Can you try:

`woeusb --device Win10.iso /dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72`

To resolve the link, we can use `readlink -f /dev/disk/by-id/foobar` - @WaxyMocha dou you want to include that step?

**Comment by kevinoid** [comment]
> What is the output of `ls -la /dev/disk/by-id/`? Does it have the `-0:0` suffix at the end?

    $ ls -la /dev/disk/by-id/
    total 0
    drwxr-xr-x 2 root root 600 Nov 28 06:45 .
    drwxr-xr-x 9 root root 180 Nov 27 23:44 ..
    lrwxrwxrwx 1 root root   9 Nov 28 06:44 ata-HL-DT-ST_DVDRAM_GT50N_M1IC2365850 -> ../../sr0
    lrwxrwxrwx 1 root root   9 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H -> ../../sda
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H-part1 -> ../../sda1
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H-part2 -> ../../sda2
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H-part3 -> ../../sda3
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H-part4 -> ../../sda4
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H-part5 -> ../../sda5
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H-part6 -> ../../sda6
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 ata-Samsung_SSD_860_EVO_1TB_S3Z8NB0K625586H-part7 -> ../../sda7
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 dm-name-sda3_crypt -> ../../dm-2
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 dm-name-sda6_crypt -> ../../dm-0
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 dm-name-sda7_crypt -> ../../dm-1
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 dm-uuid-CRYPT-LUKS1-10b111960b89486486eec48851da0c13-sda6_crypt -> ../../dm-0
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 dm-uuid-CRYPT-PLAIN-sda7_crypt -> ../../dm-1
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 dm-uuid-CRYPT-TCRYPT-sda3_crypt -> ../../dm-2
    lrwxrwxrwx 1 root root   9 Nov 28 06:45 usb-Kanguru_FlashBlu_0702353730542E72-0:0 -> ../../sdb
    lrwxrwxrwx 1 root root  10 Nov 28 06:45 usb-Kanguru_FlashBlu_0702353730542E72-0:0-part1 -> ../../sdb1
    lrwxrwxrwx 1 root root   9 Nov 28 06:45 usb-SanDisk_Cruzer_2203621B8441B9D3-0:0 -> ../../sdc
    lrwxrwxrwx 1 root root  10 Nov 28 06:45 usb-SanDisk_Cruzer_2203621B8441B9D3-0:0-part1 -> ../../sdc1
    lrwxrwxrwx 1 root root   9 Nov 28 06:44 wwn-0x5001480000000000 -> ../../sr0
    lrwxrwxrwx 1 root root   9 Nov 28 06:44 wwn-0x5002538e40524812 -> ../../sda
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 wwn-0x5002538e40524812-part1 -> ../../sda1
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 wwn-0x5002538e40524812-part2 -> ../../sda2
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 wwn-0x5002538e40524812-part3 -> ../../sda3
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 wwn-0x5002538e40524812-part4 -> ../../sda4
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 wwn-0x5002538e40524812-part5 -> ../../sda5
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 wwn-0x5002538e40524812-part6 -> ../../sda6
    lrwxrwxrwx 1 root root  10 Nov 28 06:44 wwn-0x5002538e40524812-part7 -> ../../sda7

> I have not seen that format before. Can you try:
> 
> `woeusb --device Win10.iso /dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72`

    ERROR: Target media "/dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72" is not a block device file!

**Comment by WaxyMocha** [comment]
This tool is behaving this way because the check for the device have so shaky assumptions about how the path to block device should look like.
https://github.com/WoeUSB/WoeUSB-ng/blob/master/WoeUSB/utils.py#L95

Yes, your way should work, but it is not supported.
@kevinoid do you have need for specifying device like this, or are you just reporting to us that software is just not behaving as expected? Either way, I'm sorry, but I currently lack time to do any changes to the codebase.

**Comment by JonnyTech** [comment]
@kevinoid can you try:

`woeusb --device Win10.iso /dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72-0\:0`

and

```woeusb --device Win10.iso `readlink -f /dev/disk/by-id/usb-Kanguru_FlashBlu_0702353730542E72-0\:0` ```

(the first escapes the colon `-0\:0` and the other resolves the symlink too)

**Comment by kevinoid** [comment]
Thanks @WaxyMocha!

> @kevinoid do you have need for specifying device like this, or are you just reporting to us that software is just not behaving as expected?

I did try to use the device in the example.  I find that using `/dev/disk/by-id/` reduces the chance of operating on the wrong device (since `sdb`, `sdc`, etc. change while `by-id` is stable).  However, I'm capable of working around it, so there's no hurry to fix it for my benefit.  I reported the issue because I thought it might be something this project would be interested in supporting.  Feel free to prioritize, or close, as appropriate.
